File: 1802-number-of-students-unable-to-eat-lunch/1802-number-of-students-unable-to-eat-lunch.go
func countStudents(students []int, sandwiches []int) int {
    var temp int
    var loop int
    for ;len(sandwiches)>0&&loop<=len(sandwiches); {
        if students[0] == sandwiches[0] {
            students = students[1:]
            sandwiches = sandwiches[1:]
            loop = 0
        } else {
           temp = students[0]
           students = students[1:]
           students = append(students, temp)
           loop += 1
        }
    }
    return len(students)
}

File: 1802-number-of-students-unable-to-eat-lunch/1802-number-of-students-unable-to-eat-lunch.py
import collections

class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        students_dq = collections.deque(students)
        sandwiches_dq = collections.deque(sandwiches)
        loop = 0
        while sandwiches_dq:
            if students_dq[0] == sandwiches_dq[0]:
                students_dq.popleft()
                sandwiches_dq.popleft()
                loop = 0
            else:
                temp = students_dq[0]
                students_dq.popleft()
                students_dq.append(temp)
                loop += 1
                if loop >= len(students_dq):
                    break
        return len(students_dq)
